src/snowflake/snowpark/_internal/analyzer/metadata_utils.py

```diff
@@ -103,6 +112,26 @@ def infer_metadata(
                 analyzer,
                 df_aliased_col_name_to_real_col_name,
             )
+        # When source_plan is a Join, we only infer its quoted identifiers
+        # if two plans don't have any common quoted identifier
+        # This is conservative and avoids parsing join condition, but works for Snowpark pandas,
```

Comment:
This should be a big help in cutting down on the unnecessary describe queries for pandas workloads. Great!